### DISCUSSION TITLE
Dynamic load without refresh (F5)

### DISCUSSION THREAD

**Discussion by tomaskanok** [vercel/next.js#52737] [Help]
### Summary

I have **Navbar** with links to pages **CreateUser** and **ListOfUsers**. I create new user, then I click on page ListOfUsers and he is not there. Why? It is cached until next build.

When I put _export const dynamic = 'force-dynamic'_ to page ListOfUsers, list is updated when I click refresh (F5). Thats not enough. I would like to add user, click on link in navbar and user will be in the list.

1. How to force to load users data?
2. Is this case for using some state management like Redux?
Thank you

### Additional information

```js
using nextjs 13.4.7, prisma
```

**Comment by robertpiosik** [comment]
Code written in `useEffect` hook will run in the browser, you should refetch your users there. 

Worth thinking is if you need to prerender these users, maybe it could be sufficient to only fetch them in the browser.

**Reply by robertpiosik** [reply]
You also asked about using redux, the rule of thumb is going with it when using context would cause too much bloat.

Say your app is fairly simple and all you need to access globally is the list of users - using context is absolutely fine if not encouraged.

Remember to keep your providers (e.g. providing redux store, context value) as high in the tree as possible.

**Reply by tomaskanok** [reply]
how _useEffect_ help me when users are loaded in different component? 

What I understand. I need to use hook _useContext_ in **Layout.tsx** and load users there. Then on **adduser** page add newly user to users. On page **listofusers** change fetch via useContext? But I am not able to do it, came with a lot of errors.

Sorry about my confusion. I am, comming from .net it is kind of strange behavior. I do not get it why it is so difficult. I just want to go to page listofusers and read an users from database.

sample project https://github.com/tomaskanok/test-user

BTW. it works correctly on local. When deployed to prod, page need to be refreshed to get a correct data

**Reply by robertpiosik** [reply]
What I can suggest is that your approach is prone to stale data, e.g. when there is some other source of new users, by the time you reach the list with the data returned from user creation a while ago, you would see not so fresh results.

You could mitigate it by combining your approach with some refetching of list of users data which would offer a refresh of the list if it's stale.
